### PR TITLE
Update to validate XAPI data can sent into an LRS

### PR DESCRIPTION
**Pull request by jujusb**

### CODE REVIEW

File: src/lib/activities/MinioActivity.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
 const ServerError = require('../error');
 var mongoose = require('mongoose');
 var async = require('async');
 
+// by default it creates an ideally scalable bloom filter for 8 elements with an error rate of 0.01 and a load factor of 0.5
+const filter = new ScalableBloomFilter()
+
 var Activity = require('./activity');
 
 var config = require('..//config');
```
Comment (imartinezortiz) [inline src/lib/activities/MinioActivity.js:12]: extra /

File: src/lib/activities/MinioActivity.js [hunk 2]
```diff
@@ -138,7 +143,17 @@ class MinioActivity extends Activity {
...
+					if(trace.id == null) {
+						traceid = uuidv4();
+					} else {
+						traceid = trace.id;
+					}
+					while(filter.has(traceid)) {
+						traceid = uuidv4();
+					}
+					trace.id=traceid;
+					filter.add(traceid);
```
Comment (imartinezortiz) [inline src/lib/activities/MinioActivity.js:156]: Better have function / method called generateStatementId()

File: src/lib/lti/tool.js [hunk 3]
```diff
@@ -285,7 +285,7 @@ lti.onDeepLinking(async (token, req, res) => {
 const setup = async () => {
   Log('lib/tool/lti.setup(): started');
   try{
-    await lti.deploy({ serverless: true });
+    await lti.deploy({ serverless: true, debug: true });
```
Comment (imartinezortiz) [inline src/lib/lti/tool.js:288]: debug, should be configurable from our own config

**Review by imartinezortiz — CHANGES_REQUESTED** [review]

File: src/lib/activities/MinioActivity.js (hunk 2 above)
Comment (jujusb) [reply src/lib/activities/MinioActivity.js]: fixed

File: src/lib/lti/tool.js (hunk 3 above)
Comment (jujusb) [reply src/lib/lti/tool.js]: fixed

File: src/lib/activities/MinioActivity.js (hunk 1 above)
Comment (jujusb) [reply src/lib/activities/MinioActivity.js]: fixed

**Review by imartinezortiz — APPROVED** [review]

**Pull request merged**